### ISSUE TITLE
Clearer version negotiation in connection & channel handshakes

### ISSUE THREAD

**Issue by cwgoes** [cosmos/ibc#180]
- Explicit "version" field
- Opaque to protocol in the case of channels
- Exposed to modules in the case of channels
- Explain what protocol details could be contingent on particular (tiered) version numbers

**Comment by cwgoes** [comment]
...

- should actually be negotiated, e.g. receiver can alter
- should have agreed to string by end of handshake
- should integrate ibc protocol versioning, e.g. channel datagrams can be versioned on the connection
- packets could also version, but that's out of scope of IBC/TAO

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
